control/HTTPCacheControl.php

```diff
@@ -0,0 +1,407 @@
+	 *
+	 * @see https://docs.silverstripe.org/en/developer_guides/performance/http_cache_headers/
+	 * @param bool $force Force the cache to diabled even if it's forced private or public
+	 * @return $this
+	 */
+	public function disableCache($force = false)
+	{
+		// Only execute this if its forcing level is high enough
+		if (!$this->allowChange(self::LEVEL_DISABLED, $force )) {
+			SS_Log::log("Call to disableCache($force) didn't execute as it's lower priority than a previous call", SS_Log::DEBUG);
+			return $this;
+		}
+
+		$this->state = array(
+			'no-cache' => null,
+			'no-store' => null,
+			'must-revalidate' => null,
+		);
+		return $this;
+	}
+
+	/**
+	 * Advanced way to set cache control header to a non-cacheable state.
+	 * Indicates that the response is intended for a single user and must not be stored by a shared cache.
+	 * A private cache (e.g. Web Browser) may store the response. Also removes `public` as this is a contradictory directive.
+	 *
+	 * @see https://docs.silverstripe.org/en/developer_guides/performance/http_cache_headers/
+	 * @param bool $force Force the cache to private even if it's forced public
+	 * @return $this
+	 */
+	public function privateCache($force = false)
+	{
+		// Only execute this if its forcing level is high enough
+		if (!$this->allowChange(self::LEVEL_PRIVATE, $force)) {
+			SS_Log::log("Call to privateCache($force) didn't execute as it's lower priority than a previous call", SS_Log::DEBUG);
+			return $this;
+		}
+
+		// Update the directives
+		$this->setDirective('private');
+		$this->removeDirective('public');
+		$this->removeDirective('no-cache');
+		$this->removeDirective('no-store');
+		return $this;
+	}
+
+	/**
+ 	 * Advanced way to set cache control header to a cacheable state.
+	 * Indicates that the response may be cached by any cache. (eg: CDNs, Proxies, Web browsers)
+	 * Also removes `private` as this is a contradictory directive
+	 *
+	 * @see https://docs.silverstripe.org/en/developer_guides/performance/http_cache_headers/
+	 * @param bool $force Force the cache to public even if it's private, unless it's been forced private
+	 * @return $this
+	 */
+	public function publicCache($force = false)
+	{
+		// Only execute this if its forcing level is high enough
+		if (!$this->allowChange(self::LEVEL_PUBLIC, $force)) {
+			SS_Log::log("Call to publicCache($force) didn't execute as it's lower priority than a previous call", SS_Log::DEBUG);
+			return $this;
+		}
+
+		$this->setDirective('public');
+		$this->removeDirective('private');
+		$this->removeDirective('no-cache');
+		$this->removeDirective('no-store');
+		return $this;
+	}
+
+	/**
+	 * Generate and add the `Cache-Control` header to a response object
+	 *
+	 * @param SS_HTTPResponse $response
+	 *
+	 * @return $this
+	 */
+	public function applyToResponse($response)
+	{
+		$response->addHeader('Cache-Control', $this->generateCacheHeader());
```

Comment:
Perhaps we should abstract this further into `generateHeaders`, of which `generateCacheHeader` is but one of the methods?

That way we could call `generateHeaders` in `HTTP` class, allowing our cache control service to dictate more than a single hard-coded header.

Reply:
We're not looking for further non-essential refactorings as this time. We can pick this up when it's applied to `4`, though

Reply:
Like Sam said below, let's create a nicer API for the 4.x API, but for 3.x where we don't have the option to do middleware it's fine as-is - we basically ran out of time to reach consensus that would change the status quo, so working code wins :)